JiraReporter/RestApiRequests.cs

```diff
@@ -21,19 +21,43 @@ public static T ResolveRequest<T>(Policy policy, RestRequest request, bool isXml
             else
                 client.Authenticator = new HttpBasicAuthenticator(policy.Username, policy.Password);
 
-            var results = client.Execute(request).Content;
+            return client;
+        }
+
+        public static T ResolveRequest<T>(Policy policy, RestRequest request, bool isXml = false)
+        {
+            var client = ClientLogin(policy);
+            var response = client.Execute(request);
+
+            validateResponse(response);
 
             if (isXml)
-                return Deserialization.XmlDeserialize<T>(results);
+                return Deserialization.XmlDeserialize<T>(response.Content);
             else
-                return Deserialization.JsonDeserialize<T>(results);
+                return Deserialization.JsonDeserialize<T>(response.Content);
         }
 
-        public static JiraReporter.JiraModels.Project GetProject(Policy policy)
+        public static T ResolveJiraRequest<T>(Policy policy, RestRequest request) where T : new()
+        {
+            var client = ClientLogin(policy);
+            var response = client.Execute<T>(request);
+
+            validateResponse(response);
+
+            return response.Data;
+        }
```

Comment:
Din cate m-am uitat in ultima versiune de la RestSharp [doar in mesajul de update] - ei zic ca deserializarea ar trebui sa o facem manual - recomand sa ramana asa - e conform functionalitatii AnotherJiraRestClient (ResolveJiraRequest), iar in momentul de refactor, sa facem update la packages si probabil atunci vom trece pe serializarea manuala sau trecem de tot la  Execute<T> sa faca ce face versiunea curenta (noi suntem la o versiune mai veche) in cazul in care nu sunt probleme.